frontend/interfaces/config.ts

```diff
@@ -35,6 +35,11 @@ interface ICustomSetting {
   labels_exclude_any?: string[];
 }
 
+export interface IAppleDeviceUpdates {
+  minimum_version: string | null;
+  deadline: string | null;
+}
+
```

Comment:
This interface IAppleDeviceUpdates is also needed for #19852
(API change: https://github.com/fleetdm/fleet/pull/20460/files)

Open to other naming suggestions! @jacobshandling @ghernandez345 @gillespi314

Reply:
What about `IAppleUpdateRequirement`?

Reply:
Or even just `IUpdateRequirement`? The context would be provided by the field name e.g. `macos_updates`, and this structure could apply other OSs in the future too,